### PR TITLE
DAT-15775

### PR DESCRIPTION
**Pull request by sayaliM0412**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

1. `build.yml`: removing the use of workflow-helper.js from `build.yml`
2. `run-tests.yml` : consist of `unit-tests` and `integration-tests`. If the PR is from a forked repo, only authorize will allow the workflow to run. 
3. new `build.yml` : build and publishes packages to build artifact page and GPM
4. `functional-tests` and `tests-harness` : are run after the new `build.yml` completes successfully. The link is visible on the run of run-tests.yml

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -408,12 +189,10 @@ jobs:
       # Version artifact based off of branch
       - name: Version Artifact
         run: |
-          version=${{ needs.setup.outputs.thisBranchFileName }}-SNAPSHOT
-          # Replace / with - in branch name
-          version=$(echo "$version" | sed 's/\//-/g')
+          version=${{ needs.setup.outputs.thisBranchName }}-SNAPSHOT
```
Comment (filipelautert) [inline .github/workflows/build.yml:192]: Won't we face problems in branches with / in the name as the sed was removed?

**Review by filipelautert — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (sayaliM0412) [reply .github/workflows/build.yml:192]: this is already taken care of in here -> https://github.com/liquibase/liquibase/pull/4780/files#diff-7314d0ebbd2e9537ae4889316745b4fd2fa43cb86275c9caae18a86ba228b642R79

**Review by rberezen — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,346 +1,128 @@
...
-
-
+name: Build and Publish
 
+#the concurrency check will cancel in-progress runs on non-master branches and pull requests on forked repositories, but not on the specified branches (master, github-action-**, DAT-**, dependabot-**).
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
-  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
+  cancel-in-progress: false
```
Comment (StevenMassaro) [inline .github/workflows/build.yml:6]: Is cancel-in-progress supposed to be false?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -433,35 +212,45 @@ jobs:
...
+          cp -r liquibase-standard/src/main/resources/liquibase/examples liquibase-dist/target/liquibase-${{ needs.setup.outputs.thisBranchName }}-SNAPSHOT
+          
+          ls -l liquibase-dist/target
+          #zip the above folder again which contains examples folder
+          (cd liquibase-dist/target && tar -czf liquibase-${{ needs.setup.outputs.thisBranchName }}-SNAPSHOT.tar.gz liquibase-${{ needs.setup.outputs.thisBranchName }}-SNAPSHOT)  
+          
           ## verify that the contents of the tar matches the expectation
-          echo "Comparing the contents of the tar.gz with the expected contents (using baseline file in liquibase-dist directory) to ensure that no new files were accidentally added. A failure here indicates that a new file was added to the tar.gz, and it should either be rectififed or added to the baseline."
+          #echo "Comparing the contents of the tar.gz with the expected contents (using baseline file in liquibase-dist directory) to ensure that no new files were accidentally added. A failure here indicates that a new file was added to the tar.gz, and it should either be rectififed or added to the baseline."
```
Comment (StevenMassaro) [inline .github/workflows/build.yml:230]: Suggest you uncomment this echo

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -514,173 +308,44 @@ jobs:
...
-                  "runDetails": JSON.stringify({
-                    "liquibaseBranch": helper.getCurrentBranch(),
-                    "notices": notices
-                  })
-                }
-              });
-            }
+  run-functional-tests:
+    needs: [ setup, build_publish_branch ]
+    uses: liquibase/liquibase/.github/workflows/run-functional-tests.yml@github-action-DAT-15775
```
Comment (StevenMassaro) [inline .github/workflows/build.yml:339]: Should this line have `@github-action-DAT-15775` in it?

File: .github/workflows/create-release.yml
Comment (StevenMassaro) [inline .github/workflows/create-release.yml:1]: Should the steps in this workflow be commented out?

File: .github/workflows/run-functional-tests.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+      - name: Display run ID
+        run: echo ${{ steps.return_dispatch.outputs.run_id }}
+
+      - name: Report PRO-Tests Run URL
+        uses: actions/github-script@v6.4.1
+        with:
+          script: |
+            const targetURL = "https://github.com/liquibase/liquibase-pro-tests/actions/runs/${{ steps.return_dispatch.outputs.run_id }}"
+            core.notice("Pro-Tests Run URL: " + targetURL);
```
Comment (StevenMassaro) [inline .github/workflows/run-functional-tests.yml:66]: This is a great addition, nice job!

File: .github/workflows/run-test-harness.yml
Comment (StevenMassaro) [inline .github/workflows/run-test-harness.yml:1]: This is essentially the same exact file as `.github/workflows/run-functional-tests.yml` and these could probably be combined and made more dynamic.

File: .github/workflows/build.yml (hunk 2 above)
Comment (sayaliM0412) [reply .github/workflows/build.yml:6]: I will check on this again. I remember I had to do this for testing.

File: .github/workflows/build.yml (hunk 4 above)
Comment (sayaliM0412) [reply .github/workflows/build.yml:339]: this was taken care of

File: .github/workflows/create-release.yml
Comment (sayaliM0412) [reply .github/workflows/create-release.yml:1]: The azure uber jar was creating issue during last release as well. I did not want to look into it for this ticket and go down that rabbit hole.

**Pull request merged**